tools/dev/msftidy.rb

```diff
@@ -730,6 +730,17 @@ def check_author
     end
   end
 
+  # Check for modules specifying a description
+  #
+  def check_description
+    # Payloads do not require a description
+    return if @module_type == 'payloads'
+
+    unless @source =~ /["']Description["'][[:space:]]*=>/
+      error('Missing "Description" info, please add')
+    end
+  end
```

Comment:
Looks like there's false positives here if `Description` is found within the actions definition:

https://github.com/rapid7/metasploit-framework/blob/06f84b0547b7a584d0d8387790d587216308b4c7/modules/auxiliary/analyze/apply_pot.rb#L20-L25

Reply:
I copied the regex check for author. There's probably also false positives for 'Author' too.

Reply:
False negatives rather than false positives due to requiring at least one match.

```
# git diff tools/dev/msftidy.rb
diff --git a/tools/dev/msftidy.rb b/tools/dev/msftidy.rb
index 236a51ac64..dafe35a8a5 100755
--- a/tools/dev/msftidy.rb
+++ b/tools/dev/msftidy.rb
@@ -730,6 +730,17 @@ class Msftidy
     end
   end
 
+  # Check for modules specifying a description
+  #
+  def check_description
+    # Payloads do not require a description
+    return if @module_type == 'payloads'
+
+    unless @source =~ /["']Description["'][[:space:]]*=>/
+      error('Missing "Description" info, please add')
+    end
+  end
+
   #
   # Run all the msftidy checks.
   #
@@ -765,6 +776,7 @@ class Msftidy
     check_use_datastore_debug
     check_arch
     check_author
+    check_description
   end
 
   private
# ./tools/dev/msftidy.rb modules/auxiliary/analyze/apply_pot.rb
modules/auxiliary/analyze/apply_pot.rb - [INFO] No CVE references found. Please check before you land!
```